service/src/main/java/bio/terra/workspace/service/resource/controlled/ControlledAiNotebookInstanceResource.java

```diff
@@ -146,6 +146,10 @@ public int hashCode() {
     return result;
   }
 
+  public static String generateUniqueInstanceId() {
+    return String.format("terra-%s-notebook", UUID.randomUUID());
```

Comment:
Why do notebooks need random IDs?

Agree this is necessary for buckets (which are globally namespaced), but my understanding is that you could just use the resource name here.

Reply:
ah, you are right. i also realize that the resource name is not always a valid instance id. so I made some change.